### DISCUSSION TITLE
Dictionary Help

### DISCUSSION THREAD

**Discussion by Lundo1** [community/community#21455] [Other Feature Feedback, Questions, & Ideas]
<p>I am currently going through the book “Python Crash course” by Eric Matthes, and I am having a really hard time understanding some of the code. I am teaching myself just because I think coding could be a really fun hobby. So here is what I am struggling with.</p>
<p>Chapter 6 Dictionaries and If you own the book it is favorite_languages.py on pg 102.</p>
<p>favorite_languages = {<br>
‘jen’ : ‘python’,<br>
'sarah : ‘c’,<br>
‘edward’ : ‘ruby’,<br>
‘phil’ : ‘python’,<br>
}<br>
friends = [‘phil’, ‘sarah’]<br>
for name in favorite_languages.keys():<br>
print(f"Hi {name.title()}.")</p>
<pre><code>if name in friends:
       language = favorite_languages[name].title()
       print(f"\t{name.title()}, I see you love {language}!")
</code></pre>
<p>We get the output</p>
<p>Hi Jen.<br>
Hi Sarah.<br>
Sarah, I see you love C!<br>
Hi Edward.<br>
Hi Phil.<br>
Phil, I see you love Python!</p>
<p>I understand what they are trying to do, but my brain does not understand the coorelations, and why we are getting that output. So i was hoping someone could break this down for me and explain exactly whats going on.</p>
<p>Thanks</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hey <a class="mention" href="/lundo1">@Lundo1</a> and welcome to the GitHub Community Forum! Yes, coding could definitely be a fun hobby and everyone can learn! Let me break down the code step-by-step for you.</p>
<pre><code class="lang-python"># Below here, a dictionary is defined. Just like in real life, a
# dictionary in Python maps so-called "keys" (in real-life the words,
# but in this case some names) to "values" (in real-life the
# definitions, but in this case their favorite programming language).
favorite_languages = {
    'jen' : 'python',
    'sarah' : 'c',
    'edward' : 'ruby',
    'phil' : 'python’'
}

# Here, we create a list that contains names of your friends:
friends = ['phil', 'sarah']

# Here we make a for-loop. We loop through the keys, since the keys()
# method of a dictionary returns a list of the keys. In this case:
# favorite_languages.keys() = ['jen', 'sarah', 'edward', 'phil'].
# Similarly, you could also use .values() for the items, or .items()
# for both the key and the value, like:
# &gt; for name, fav_language in favorite_languages.items():
# &gt;     ...
for name in favorite_languages.keys():

    # Using an f-string we print the name. See what is going
    # on in an f-string?
    print(f'Hi {name.title()}.')

    if name in friends:  # If the current name is in the friends list

        # Determine their favorite language. Using square
        # brackets ( [ and ] ) you can access values from a
        # dictionary. In your case: favorite_languages['phil'] == 'python'.
        # The .title() may be a bit confusing, but that method just capitalizes
        # the first letter of the programming language. Leave it out and test
        # what happens!
        language = favorite_languages[name].title()

        # Again, print some info using an f-string.
        print(f'\t{name.title()}, I see you love {language}!')
</code></pre>
<p>Hope it helps! If you have any further questions, let us know!</p>

**Comment by Lundo1** [comment]
<p>Thanks for the reply. Sorry I didn’t get back to you sooner, work has been crazy. Really appreciate your help!</p>
